ui/components/component-library/text/text.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { forwardRef, Ref } from 'react';
+import classnames from 'classnames';
+import Box from '../../ui/box';
+import {
+  FONT_WEIGHT,
+  TextVariant,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { TextProps } from './text.types';
+
+const getTextElementDefault = (variant: TextVariant) => {
+  switch (variant) {
+    case TextVariant.displayMd:
+      return 'h1';
+    case TextVariant.headingLg:
+      return 'h2';
+    case TextVariant.headingMd:
+      return 'h3';
+    case TextVariant.headingSm:
+      return 'h4';
+    case TextVariant.inherit:
+      return 'span';
+    // TextVariant.bodyLgMedium, TextVariant.bodyMd, TextVariant.bodyMdBold, TextVariant.bodySm, TextVariant.bodySmBold, TextVariant.bodyXs use default 'p' tag
+    default:
+      return 'p';
+  }
+};
+
+export const Text = forwardRef(function Text(
+  {
+    variant = TextVariant.bodyMd,
+    color = TextColor.textDefault,
+    fontWeight,
+    fontStyle,
+    textTransform,
+    textAlign,
+    textDirection,
+    overflowWrap,
+    ellipsis,
+    as,
+    className = '',
+    children,
+    ...props
+  }: TextProps,
+  ref: Ref<HTMLDivElement>,
```

Comment:
Should this be any HTML element? `h1-h6` `p`, `li` etc
```suggestion
  ref: Ref<HTMLElement>,
```